server/src/models/users.js

```diff
@@ -94,6 +97,35 @@ const userSchema = new mongoose.Schema({
     timestamps: true
 })
 
+userSchema.statics.checkCredentials = async (email, password) => {
+    const user = await User.findOne({
+        email
+    })
+
+    if (!user) {
+        throw new Error('Invalid operation')
```

Comment:
I don't think this provides any useful message to the user? Consider giving a message like 'Invalid Credentials'

Reply:
Also, where is the error thrown being handled?

Reply:
Sure (for the first part)
And for the second part, the error will be handled where this function was called. Since this function is asynchronous, it must be called by using an async await or by using a promise and hence by throwing an error, we are basically rejecting the promise.